From 4a555ffb77a5947814b6c7f330968318e265c496 Mon Sep 17 00:00:00 2001
From: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date: Wed, 28 Sep 2022 21:17:17 +0200
Subject: [PATCH] Makefile: move 'EXTRA_CFLAGS += $(USER_EXTRA_CFLAGS)' at the
 end of EXTRA_FLAGS assignment

At the moment USER_EXTRA_CFLAGS can't override local Makfile EXTRA_CFLAGS
since it's assigned at the beginning of the Makefile. For example it's not
possible to undefine the hardcoded CONFIG_LITTLE_ENDIAN and this doesn't
allow to build these modules for big endian architectures. So let's move
the assignment of USER_EXTRA_CFLAGS to EXTRA_CFLAGS after the last
EXTRA_CFLAGS assignment.

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
[Upstream status: https://github.com/jwrdegoede/rtl8189ES_linux/pull/83]
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8fcb7bd..e4664e9 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,3 @@
-EXTRA_CFLAGS += $(USER_EXTRA_CFLAGS)
 EXTRA_CFLAGS += -O1
 #EXTRA_CFLAGS += -O3
 #EXTRA_CFLAGS += -Wall
@@ -2239,6 +2238,8 @@ ifneq ($(USER_MODULE_NAME),)
 MODULE_NAME := $(USER_MODULE_NAME)
 endif
 
+EXTRA_CFLAGS += $(USER_EXTRA_CFLAGS)
+
 ifneq ($(KERNELRELEASE),)
 
 ########### this part for *.mk ############################
-- 
2.34.1

